fuel/src/main/kotlin/com/github/kittinunf/fuel/util/Base64.kt

```diff
@@ -0,0 +1,80 @@
+/*
+Copied From: https://gist.github.com/hrules6872/e2d4d02a1e8d3c6328ae5aeabc430b96
+ */
+package com.github.kittinunf.fuel.util
+
+import java.io.ByteArrayOutputStream
+
+fun String.encodeBase64ToString(): String = String(this.toByteArray().encodeBase64())
+fun String.encodeBase64ToByteArray(): ByteArray = this.toByteArray().encodeBase64()
+fun ByteArray.encodeBase64ToString(): String = String(this.encodeBase64())
+
+fun String.decodeBase64(): String = String(this.toByteArray().decodeBase64())
+fun String.decodeBase64ToByteArray(): ByteArray = this.toByteArray().decodeBase64()
+fun ByteArray.decodeBase64ToString(): String = String(this.decodeBase64())
+
+fun ByteArray.encodeBase64(): ByteArray {
+    val table = (CharRange('A', 'Z') + CharRange('a', 'z') + CharRange('0', '9') + '+' + '/').toCharArray()
+    val output = ByteArrayOutputStream()
+    var padding = 0
+    var position = 0
+    while (position < this.size) {
+        var b = this[position].toInt() and 0xFF shl 16 and 0xFFFFFF
+        if (position + 1 < this.size) b = b or (this[position + 1].toInt() and 0xFF shl 8) else padding++
+        if (position + 2 < this.size) b = b or (this[position + 2].toInt() and 0xFF) else padding++
```

Comment:
would this be clearer if this was written as 
```kotlin 
b = if (position + 2 < this.size) b or (this[position + 2].toInt() and 0xFF) else padding++
```

Reply:
I see what you meant.